### PR TITLE
WIP like feature for posts

### PR DESCRIPTION
**Pull request by yoongkang**
Hi, I've been looking at #541 and started some preliminary hacking. Before I go much further I'd like to open a discussion with regards to the specifics of the implementations (design, categories of 'likes', etc.), and also do a code review.

I have a module called "Likeable" which you can now just include in any model and it will become a likeable model. Should be extensible to photos, crops, etc.

At the moment I've only added it to posts, and it's still ugly as. No AJAX yet either. 

Hopefully I'll get to drop by one of the hack nights so we can iron this out.

### CODE REVIEW

File: app/models/likeable.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Likeable
+  extend ActiveSupport::Concern
```
Comment (Skud) [inline app/models/likeable.rb]: Love that we're doing this with concerns. I'd love to make our photos code be a concern too, in future.

**Comment by Skud** [comment]
This looks fantastic. I particularly love the use of concerns (which I think we should extend to photos, too, in future).

Agreed that we should talk about design (in the visual sense) and categories for likes.  I think I should raise the latter, at least, on Growstuff Talk.  I'll start a thread there.

**Comment by tygriffin** [comment]
Like!

**Comment by cesy** [comment]
I'd love to get this moving again. @yoongkang, are you still keen for hacking on it? We've got a great start here. The other thing mentioned in the issue is having reasons for liking a post, and for those reasons to be configurable by an admin, I think, but that's not necessarily required for minimum viable product and can be added in a future release. Similarly, we can spin off a separate issue for AJAX-ifying it.

So what else is needed to make this usable? Next steps are to fix the merge conflicts and share a couple of screenshots for input on the design, I think, to see how much CSS change is needed before we go live with it.

**Comment by yoongkang** [comment]
@cesy Yep, I'm happy to continue with this. Hopefully I'll have a bit of time this weekend to hack on it. 

If you're happy right now to do a page reload every time a 'Like' button is clicked, I'll just have to make it not ugly and it should be good to go for another review.

AJAX will require a unique id for every element containing the Like link for lists of likeable items. I can't remember what stopped me from implementing that but I'll take a look again.

**Comment by cesy** [comment]
Have you had a chance to look at this, or are you interested in picking it up again? We've just started deploying more often again.

File: app/models/likeable.rb (hunk 1 above)
Comment (maco) [inline app/models/likeable.rb]: I think Rails convention is to put model concerns in app/models/concerns/

**Comment by maco** [comment]
A "type" field on the Like would be a good thing to pop in. The full feature includes having many types of reactions (hmm Facebook is calling them "reactions" now that they added things other than just "like" and that might be a better class name), and if we create `ReactionType` or `LikeType` now we can avoid having to have a migration later that shifts all uncategorized Likes into a category.

**Comment by yoongkang** [comment]
Sorry, I've been way too busy. Happy to free it up for anyone else who's interested.

**Pull request closed without merging**